up-l1/README.adoc

```diff
@@ -236,18 +205,16 @@ receive(
 | The _sink_ address pattern that messages need to match. If omitted, a message must not contain any sink address in order to match.
 
 | result
-| Future<Void, link:../basics/error_model.adoc[UStatus]>
+| Future<UMessage>
 | A succeeded future containing the least recent message that matches the given filter criteria and has not expired yet. +
-A failed future if no message can be retrieved from the messaging infrastructure. The `UStatus` will contain a `UCode` indicating the reason for the failure.
+Otherwise, a failed future which contains a `UCode` indicating the reason for the failure.
 
 |===
 
-[oft-sid="req~utransport-register-listener~1"]
---
 Implementations
 
+* *MAY* fail invocations with a `UCode::UNIMPLEMENTED` if the transport does not support the _pull_ <<_delivery_method>>
 * *MUST* fail with a `UCode::NOT_FOUND` if there are no matching messages available
---
 
 === Message Delivery
 
```

Comment:
Can't quite leave a comment that far down, so:

```
* Transport *MUST* support retransmission of messages that are no able to be delivered to the next-hop
```
=>
```
* Transport *MUST* support retransmission of messages that are not able to be delivered to the next-hop
```

Reply:
That whole section needs to be rewritten FMPOV. I'll give it a shot ...